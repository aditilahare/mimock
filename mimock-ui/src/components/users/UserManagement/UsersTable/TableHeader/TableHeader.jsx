import React from 'react';
import { TableHeaderWrapper, HeaderItem } from './TableHeader.style';

export default function TableHeader() {
	const headers = [
		{ label: 'Name', key: 'name' },
		{ label: 'User name', key: 'userName' },
		{ label: 'Role', key: 'role' },
		{ label: 'Activation status', key: 'activationStatus' },
		{ label: 'Created on', key: 'createdOn' },
		{ label: 'Actions', key: 'actions' },
	];

	return (
		<TableHeaderWrapper data-testid='user-mgmt-table-header'>
			<For each='header' of={headers}>
				<HeaderItem data-testid={`header-item-${header.key}`} key={header.key}>
					{header.label}
				</HeaderItem>
			</For>
		</TableHeaderWrapper>
	);
}
