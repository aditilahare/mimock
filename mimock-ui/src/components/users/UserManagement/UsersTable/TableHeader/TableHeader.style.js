import tw from 'tailwind-styled-components';

export const TableHeaderWrapper = tw.div`
    w-11/12
    flex
    flex-row
    justify-between
    p-4
    bg-gray-100
    border
    border-gray-300
    rounded-md
    shadow-sm
    mx-auto
    items-center
    align-middle
    my-1
`;

export const HeaderItem = tw.div`
    font-sans
    font-bold
    text-center
    antialiased 
    text-gray-600
    w-1/4
`;
