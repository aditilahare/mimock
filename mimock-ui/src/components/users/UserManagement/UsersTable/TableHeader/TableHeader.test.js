import React from 'react';
import { render } from '@testing-library/react';
import TableHeader from './TableHeader';

describe('TableHeader', () => {
	it('should render user list table header', async () => {
		const { container, getByTestId } = await render(<TableHeader />);

		expect(getByTestId('user-mgmt-table-header')).toBeInTheDocument();

		expect(getByTestId('header-item-name')).toBeInTheDocument();
		expect(getByTestId('header-item-userName')).toBeInTheDocument();
		expect(getByTestId('header-item-role')).toBeInTheDocument();
		expect(getByTestId('header-item-activationStatus')).toBeInTheDocument();
		expect(getByTestId('header-item-createdOn')).toBeInTheDocument();
		expect(getByTestId('header-item-actions')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
