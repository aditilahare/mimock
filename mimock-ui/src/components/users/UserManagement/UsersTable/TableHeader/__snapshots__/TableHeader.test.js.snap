// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader should render user list table header 1`] = `
<div>
  <div
    class="w-11/12 flex flex-row justify-between p-4 bg-gray-100 border border-gray-300 rounded-md shadow-sm mx-auto items-center align-middle my-1"
    data-testid="user-mgmt-table-header"
  >
    <div
      class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
      data-testid="header-item-name"
    >
      Name
    </div>
    <div
      class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
      data-testid="header-item-userName"
    >
      User name
    </div>
    <div
      class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
      data-testid="header-item-role"
    >
      Role
    </div>
    <div
      class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
      data-testid="header-item-activationStatus"
    >
      Activation status
    </div>
    <div
      class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
      data-testid="header-item-createdOn"
    >
      Created on
    </div>
    <div
      class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
      data-testid="header-item-actions"
    >
      Actions
    </div>
  </div>
</div>
`;
